src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -2203,7 +2438,93 @@ private function step_in_column_group(): bool {
 	 * @return bool Whether an element was found.
 	 */
 	private function step_in_table_body(): bool {
-		$this->bail( 'No support for parsing in the ' . WP_HTML_Processor_State::INSERTION_MODE_IN_TABLE_BODY . ' state.' );
+		$tag_name = $this->get_tag();
+		$op_sigil = $this->is_tag_closer() ? '-' : '+';
+		$op       = "{$op_sigil}{$tag_name}";
+
+		switch ( $op ) {
+			/*
+			 * > A start tag whose tag name is "tr"
+			 */
+			case '+TR':
+				$this->state->stack_of_open_elements->clear_to_table_body_context();
+				$this->insert_html_element( $this->state->current_token );
+				$this->state->insertion_mode = WP_HTML_Processor_State::INSERTION_MODE_IN_ROW;
+				return true;
+
+			/*
+			 * > A start tag whose tag name is one of: "th", "td"
+			 */
+			case '+TH':
+			case '+TD':
+				// @todo Indicate a parse error once it's possible.
+				$this->state->stack_of_open_elements->clear_to_table_body_context();
+				$this->insert_html_element(
+					new WP_HTML_Token( null, 'TR', false )
+				);
+				$this->state->insertion_mode = WP_HTML_Processor_State::INSERTION_MODE_IN_ROW;
+				return $this->step( self::REPROCESS_CURRENT_NODE );
+
+			/*
+			 * > An end tag whose tag name is one of: "tbody", "tfoot", "thead"
+			 */
+			case '-TBODY':
+			case '-TFOOT':
+			case '-THEAD':
+				if (
+					! $this->state->stack_of_open_elements->has_element_in_table_scope( $tag_name )
+				) {
+					// @todo Indicate a parse error once it's possible.
+					return $this->step();
+				}
+				$this->state->stack_of_open_elements->pop();
+				$this->state->insertion_mode = WP_HTML_Processor_State::INSERTION_MODE_IN_TABLE;
+				return true;
+
+			/*
+			 * > A start tag whose tag name is one of: "caption", "col", "colgroup", "tbody", "tfoot", "thead"
+			 * > An end tag whose tag name is "table"
+			 */
+			case '+CAPTION':
+			case '+COL':
+			case '+COLGROUP':
+			case '+TBODY':
+			case '+TFOOT':
+			case '+THEAD':
+			case '-TABLE':
+				if (
+					! $this->state->stack_of_open_elements->has_element_in_table_scope( 'TBODY' ) &&
+					! $this->state->stack_of_open_elements->has_element_in_table_scope( 'THEAD' ) &&
+					! $this->state->stack_of_open_elements->has_element_in_table_scope( 'TFOOT' )
+				) {
+					// @todo Indicate a parse error once it's possible.
```

Comment:
just curious about this, because these elements are implicitly created on `+TABLE`, so I think the only case we could get here is with a TEMPLATE

```html
<table>
<td><template>This </table> does not close the outer one</template>
</table>
```

Reply:
This is just following the spec (quoted below). I'm reluctant to deviate from the spec in all but the most obvious places. There are many places the switch modes and reprocess tokens _or_ process tokens following the rules for other modes.

> A character token, if the [current node](https://html.spec.whatwg.org/multipage/parsing.html#current-node) is [table](https://html.spec.whatwg.org/multipage/tables.html#the-table-element), [tbody](https://html.spec.whatwg.org/multipage/tables.html#the-tbody-element), [template](https://html.spec.whatwg.org/multipage/scripting.html#the-template-element), [tfoot](https://html.spec.whatwg.org/multipage/tables.html#the-tfoot-element), [thead](https://html.spec.whatwg.org/multipage/tables.html#the-thead-element), or [tr](https://html.spec.whatwg.org/multipage/tables.html#the-tr-element) element
> Let the pending table character tokens be an empty list of tokens.
> 
> Let the [original insertion mode](https://html.spec.whatwg.org/multipage/parsing.html#original-insertion-mode) be the current [insertion mode](https://html.spec.whatwg.org/multipage/parsing.html#insertion-mode).
> 
> Switch the [insertion mode](https://html.spec.whatwg.org/multipage/parsing.html#insertion-mode) to "[in table text](https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-intabletext)" and reprocess the token.

Reply:
there was no hint of asking you to deviate from the spec; all I was doing was observing an interesting oddity since the only elements we're checking are ones that should always be open. I think that `TEMPLATE` is the only exception. this was just exploring the rules.

if this turns out to be true, it might be possible one day to verify and take advantage of. for now, all it did was seem odd when I saw it, and then I realized there's an escape hatch in almost all of these situations - the `TEMPLATE` element.